### PR TITLE
Add limit mounts tests

### PR DESCRIPTION
**Pull request by maryvictol**
This PR provides tests for default mount limits on a per-user basis #2210

### CODE REVIEW

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/LimitMountsTest.java [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+import com.epam.pipeline.autotests.ao.ToolTab;
+import com.epam.pipeline.autotests.mixins.Authorization;
+import com.epam.pipeline.autotests.mixins.Navigation;
+import com.epam.pipeline.autotests.mixins.Tools;
+import com.epam.pipeline.autotests.utils.C;
+import com.epam.pipeline.autotests.utils.TestCase;
+import com.epam.pipeline.autotests.utils.Utils;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/LimitMountsTest.java]: Clean up unused imports, please

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/LimitMountsTest.java [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+import static com.epam.pipeline.autotests.utils.Privilege.WRITE;
+import static com.epam.pipeline.autotests.utils.PrivilegeValue.ALLOW;
+import static java.lang.String.format;
+
+public class LimitMountsTest extends AbstractSeveralPipelineRunningTest implements Navigation, Authorization, Tools {
+
+    private String storage1 = "limitMountsStorage" + Utils.randomSuffix();
+    private String storage2 = "limitMountsStorage" + Utils.randomSuffix();
+    private String storage3 = "limitMountsStorage" + Utils.randomSuffix();
+    private String storage4 = "limitMountsStorage" + Utils.randomSuffix();
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/LimitMountsTest.java]: May be 'final'

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/LimitMountsTest.java [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+        library()
+                .removeStorageIfExists(storage1)
+                .removeStorageIfExists(storage2)
+                .removeStorageIfExists(storage3)
+                .removeStorageIfExists(storage4);
+        cleanToolLimitMounts();
+        cleanUserLimitMounts();
+    }
+
+    @Test(priority = 0)
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/LimitMountsTest.java]: Remove redundant parameter

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/LimitMountsTest.java [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+        navigationMenu()
+                .settings()
+                .switchToMyProfile()
+                .doNotMountStoragesSelect(true);
+        tools()
+                .perform(registry, group, testTool, tool ->
+                    tool.run(this))
+                .showLog(getLastRunId())
+                .expandTab(PARAMETERS)
+                .ensure(configurationParameter("CP_CAP_LIMIT_MOUNTS", "None"), exist)
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/LimitMountsTest.java]: "CP_CAP_LIMIT_MOUNTS" -> extract to constant, please

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/LimitMountsTest.java [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+                .createStorage(storage1)
+                .createStorage(storage2)
+                .createStorage(storage3)
+                .clickOnCreateStorageButton()
+                .setStoragePath(storage4)
+                .clickSensitiveStorageCheckbox()
+                .ok();
+        givePermissionsToStorage(user, storage2);
+        givePermissionsToStorage(user, storage3);
+        givePermissionsToStorage(user, storage4);
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/LimitMountsTest.java]: Could be changed to `Stream.of(storage2, storage3, storage4).forEach(s -> givePermissionsToStorage(user, s));`

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/SettingsPageAO.java [hunk 6]
```diff
@@ -1671,13 +1680,39 @@ public SystemLogsAO setIncludeServiceAccountEventsOption() {
...
         public MyProfileAO validateUserName(String user) {
             return ensure(USER_NAME, text(user));
         }
 
+        public SelectLimitMountsPopupAO<MyProfileAO> limitMountsPerUser() {
+            click(LIMIT_MOUNTS);
+            return new SelectLimitMountsPopupAO<>(this).sleep(2, SECONDS);
+        }
+
+        public MyProfileAO doNotMountStoragesSelect (boolean isSelected) {
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/SettingsPageAO.java]: Remove a space between method name and parameter, please

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/SettingsPageAO.java [hunk 7]
```diff
@@ -1671,13 +1680,39 @@ public SystemLogsAO setIncludeServiceAccountEventsOption() {
...
+            if ((!get(DO_NOT_MOUNT_STORAGES).has(cssClass("ant-checkbox-checked")) && isSelected) ||
+                    (get(DO_NOT_MOUNT_STORAGES).has(cssClass("ant-checkbox-checked")) && !isSelected)) {
+                click(DO_NOT_MOUNT_STORAGES);
+                sleep(1, SECONDS);
+            }
+            return this;
+        }
+
+        public MyProfileAO assertDoNotMountStoragesIsChecked() {
```
Comment (kamyshova) [inline e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/SettingsPageAO.java]: The method is not used. Are we sure want to keep it?

**Review by kamyshova — CHANGES_REQUESTED** [review]

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/LimitMountsTest.java (hunk 1 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/LimitMountsTest.java]: Done

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/LimitMountsTest.java (hunk 2 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/LimitMountsTest.java]: Done

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/LimitMountsTest.java (hunk 3 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/LimitMountsTest.java]: Done

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/LimitMountsTest.java (hunk 4 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/LimitMountsTest.java]: Done

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/LimitMountsTest.java (hunk 5 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/LimitMountsTest.java]: Done

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/SettingsPageAO.java (hunk 6 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/SettingsPageAO.java]: Done

File: e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/SettingsPageAO.java (hunk 7 above)
Comment (maryvictol) [reply e2e/gui/src/test/java/com/epam/pipeline/autotests/ao/SettingsPageAO.java]: Method has been removed

**Review by kamyshova — APPROVED** [review]

**Pull request merged**